Source/Core/InputCommon/ControllerInterface/SDL/SDLGamepad.cpp

```diff
@@ -293,38 +301,38 @@ std::string GameController::Axis::GetName() const
 
 ControlState GameController::Button::GetState() const
 {
-  return SDL_GameControllerGetButton(m_gc, m_button);
+  return SDL_GetGamepadButton(m_gc, m_binding.output.button);
 }
 
 ControlState GameController::Axis::GetState() const
 {
-  return ControlState(SDL_GameControllerGetAxis(m_gc, m_axis)) / m_range;
+  return ControlState(SDL_GetGamepadAxis(m_gc, m_axis)) / m_range;
 }
 
 bool GameController::Button::IsMatchingName(std::string_view name) const
 {
   if (GetName() == name)
     return true;
 
-  // So that SDL can be a superset of XInput
-  if (name == "Button A")
-    return GetName() == "Button S";
-  if (name == "Button B")
-    return GetName() == "Button E";
-  if (name == "Button X")
-    return GetName() == "Button W";
-  if (name == "Button Y")
-    return GetName() == "Button N";
-
-  // Match legacy names.
-  const auto bind = SDL_GameControllerGetBindForButton(m_gc, m_button);
-  switch (bind.bindType)
+  // We display cardinal NESW button names but also match the label name for old configs.
```

Comment:
This does not match the previous behavior, which was positional (despite using the labels in the names).
Tested on a Switch Pro Controller: before, "Button A" matched with the South button, but with this change it matches with the East button.
I don't have a PlayStation controller, but I suppose that there the XInput configs won't work at all, since `SDL_GetGamepadButtonLabel` will return Circle, Square, etc.

(Also, this part isn't for "old configs", it's for compatibility with XInput configs.)